repo/maintenancestats/types.go

```diff
@@ -0,0 +1,449 @@
+func (cs *CleanupSupersededIndexesStats) Summary() string {
+	return fmt.Sprintf("Cleaned up %v(%v) superseded index blobs", cs.DeletedBlobs, cs.DeletedSize)
+}
+
+// Type returns the type name for CleanupSupersededIndexesStats
+func (cs *CleanupSupersededIndexesStats) Type() string {
+	return cleanupSupersededIndexesStatsType
+}
+
+const generateRangeCheckpointStatsType = "generateRangeCheckpointStats"
+
+// GenerateRangeCheckpointStats delivers the statistics for MaybeGenerateRangeCheckpoint
+type GenerateRangeCheckpointStats struct {
+	FirstEpoch uint32 `json:"firstEpoch"`
+	LastEpoch  uint32 `json:"lastEpoch"`
+}
+
+// WriteValueTo writes GenerateRangeCheckpointStats to JSONWriter
+func (gs *GenerateRangeCheckpointStats) WriteValueTo(jw *contentlog.JSONWriter) {
+	jw.BeginObjectField(gs.Type())
+	jw.UInt32Field("firstEpoch", gs.FirstEpoch)
+	jw.UInt32Field("lastEpoch", gs.LastEpoch)
+	jw.EndObject()
+}
+
+// Summary generates readable summary for GenerateRangeCheckpointStats
+func (gs *GenerateRangeCheckpointStats) Summary() string {
+	return fmt.Sprintf("Generated a range checkpoint from epoch %v to %v", gs.FirstEpoch, gs.LastEpoch)
+}
+
+// Type returns the type name for GenerateRangeCheckpointStats
+func (gs *GenerateRangeCheckpointStats) Type() string {
+	return generateRangeCheckpointStatsType
+}
+
+const advanceEpochStatsType = "advanceEpochStats"
+
+// AdvanceEpochStats delivers the statistics for MaybeAdvanceWriteEpoch
+type AdvanceEpochStats struct {
+	CurEpoch uint32 `json:"curEpoch"`
+	Advanced bool   `json:"advanced"`
+}
+
+// WriteValueTo writes AdvanceEpochStats to JSONWriter
+func (as *AdvanceEpochStats) WriteValueTo(jw *contentlog.JSONWriter) {
+	jw.BeginObjectField(as.Type())
+	jw.UInt32Field("curEpoch", as.CurEpoch)
+	jw.BoolField("advanced", as.Advanced)
+	jw.EndObject()
+}
+
+// Summary generates readable summary for AdvanceEpochStats
+func (as *AdvanceEpochStats) Summary() string {
+	var message string
+	if as.Advanced {
+		message = fmt.Sprintf("Advanced epoch to %v", as.CurEpoch+1)
+	} else {
+		message = fmt.Sprintf("Stay at epoch %v", as.CurEpoch)
+	}
+
+	return message
+}
+
+// Type returns the type name for AdvanceEpochStats
+func (as *AdvanceEpochStats) Type() string {
+	return advanceEpochStatsType
+}
+
+const compactSingleEpochStatsType = "compactSingleEpochStats"
+
+// CompactSingleEpochStats delivers the statistics for MaybeCompactSingleEpoch
+type CompactSingleEpochStats struct {
+	BlobCount uint32 `json:"blobCount"`
+	BlobSize  int64  `json:"blobSize"`
+	Epoch     uint32 `json:"epoch"`
+}
+
+// WriteValueTo writes CompactSingleEpochStats to JSONWriter
+func (cs *CompactSingleEpochStats) WriteValueTo(jw *contentlog.JSONWriter) {
+	jw.BeginObjectField(cs.Type())
+	jw.UInt32Field("blobCount", cs.BlobCount)
+	jw.Int64Field("blobSize", cs.BlobSize)
+	jw.UInt32Field("epoch", cs.Epoch)
+	jw.EndObject()
+}
+
+// Summary generates readable summary for CompactSingleEpochStats
```

Comment:
No need to repeat the type in the comment, it is implied from the receiver type.
```suggestion
// Summary generates a human readable summary for the stats
```